File: 플젝1b/source/src/InstTable.java
import java.util.HashMap;
import java.io.*;
import java.util.*;

//명령어 정보를 가진 클래스이다.
public class InstTable {
	HashMap<String, Instruction> instMap;
	
	public InstTable(String instFile) {
		instMap = new HashMap<String, Instruction>();
		openFile(instFile);
	}
	
	//라인 한 줄을 통해 명령어 정보를 뽑아 저장한다.
	public void openFile(String fileName) {
		try{
			File input = new File(fileName);
			Scanner temp;
			temp = new Scanner(input);
			
			while(temp.hasNextLine()) {
				String line = temp.nextLine();
				
				Instruction temp_inst = new Instruction(line);
				instMap.put(temp_inst.name,temp_inst);
			}
			temp.close();
		}
		catch(FileNotFoundException e){
			System.out.println("file not found");
			e.printStackTrace();
		}
	}


}

class Instruction {
	String name;
	int opcode;
	int format;
	int operandN;

	//생성과 동시에 파싱한다.
	public Instruction(String line) {
		parsing(line);
	}
	
	//탭키를 구분자로 설정한다.
	public void parsing(String line) {
		Scanner temp_line = new Scanner(line).useDelimiter("\t");
		name = temp_line.next();
		opcode = Integer.parseInt(temp_line.next(),16);
		format = temp_line.nextInt();
		operandN = temp_line.nextInt();
	}

}

File: 플젝1c/source/Assembler_python/InstTable.py
#InstTable.py
class InstTable:
    def __init__(self,instFile):
        self.instMap={}
        self.openFile(instFile)

    def openFile(self,fileName):
        try:
            f = open(fileName)
            while True:
                line = f.readline()
                if not line :
                    break
                temp = Instruction(line)
                self.instMap[temp.name] = temp
        except:
            print('파일로드 실패')
        finally:
            f.close()

class Instruction:
    def __init__(self,line):
        self.parsing(line)

    def parsing(self,line):
        (self.name,self.opcode,temp_format,self.operandN) = line.strip().split('\t')
        self.format = int(temp_format)